l10n_es_aeat_mod349/models/mod349.py

```diff
@@ -215,11 +215,43 @@ def _create_349_refund_records(self):
             if original_details:
                 # There's at least one previous 349 declaration report
                 report = original_details.mapped('report_id')[:1]
-                original_details = original_details.filtered(
-                    lambda d: d.report_id == report)
-                origin_amount = sum(original_details.mapped('amount_untaxed'))
+                partner_id = original_details.mapped('partner_id')[:1]
                 period_type = report.period_type
                 year = str(report.year)
+
+                key_vals = data.get((partner, op_key, period_type, year))
+                if key_vals:
+                    key_vals['refund_details'] += refund_details
+                    continue
+
+                # Sum all details period origin
+                all_details_period = detail_obj.search([
+                    ('partner_id', '=', partner_id.id),
+                    ('partner_record_id.operation_key', '=', op_key),
+                    ('report_id', '=', report.id),
+                ], order='report_id desc')
+                origin_amount = sum(
+                    all_details_period.mapped('amount_untaxed'))
+
+                # If there are intermediate periods between the original
+                # period and the period where the rectification is taking
+                # place, it's necessary to check if there is any rectification
+                # of the original period in between these periods. This
+                # happens in this way because the right original_amount
+                # will be the value of the total_operation_amount
+                # corresponding to the last period found in between the periods
+                other_invoice_period = all_details_period.mapped(
+                    'invoice_id') - origin_invoice
+                if other_invoice_period.mapped('refund_invoice_ids'):
+                    last_refund_detail = refund_detail_obj.search([
+                        ('report_id.date_start', '>', report.date_end),
+                        ('report_id.date_end', '<', self.date_start),
+                        ('invoice_id', 'in', other_invoice_period.mapped(
+                            'refund_invoice_ids').ids)
+                    ], order='date desc', limit=1)
```

Comment:
En el log de odoo aparece el siguiente mensaje:
`WARNING test odoo.models: Model 'l10n.es.aeat.mod349.partner_refund_detail' cannot be sorted on field 'date' (not a column)`
Se debe a que el campo es un `related` y no se encuentra en base de datos. [sugerencia](https://github.com/OCA/l10n-spain/pull/1641/commits/3ea680882e3dd93193d361e3b4a9e168ee45a1f7)

Reply:
Buenas @miguels73, voy a ver si saco tiempo esta semana y reviso tu sugerencia, muchas gracias